src/openklant/config/models.py

```diff
@@ -0,0 +1,50 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from solo.models import SingletonModel
+from zgw_consumers.models import Service
+
+from openklant.components.klantinteracties.models import Klantcontact
+from referentielijsten_client.client import ReferentielijstenClient
+
+
+class ReferentielijstenConfig(SingletonModel):
+    enabled = models.BooleanField(default=False)
+    service = models.ForeignKey(
+        Service,
+        on_delete=models.PROTECT,
+        null=True,
+        blank=True,
+        help_text="The service to fetch referentielijst data from.",
+    )
+    tabel_code = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text="Code of the table containing kanaal options.",
+    )
+
+    def clean(self):
+        if not self.enabled:
+            return
+
+        if not self.service or not self.tabel_code:
+            raise ValidationError(
+                "Service and tabel_code must be set when validation is enabled."
+            )
+
+        client = ReferentielijstenClient(service=self.service)
+        items = client.get_items_by_tabel_code(self.tabel_code)
+        valid_kanalen = {item.get("naam") for item in items if item.get("naam")}
```

Comment:
```suggestion
        valid_kanalen = {item.get("code") for item in items if item.get("code")}
```
Is better to check the code, which is unique, i think also in the `KanaalValidator` you check the `code`, right?

Reply:
yes